ProjFS.Mac/PrjFSKext/KextLog.hpp

```diff
@@ -110,7 +110,7 @@ template <typename... args>
 
 #define KextLog_VnodeOp(vnode, vnodeType, procname, action, message) \
     do { \
-        if (VDIR == vnodeType) \
+        if (VDIR == (vnodeType)) \
```

Comment:
Are these ()'s needed?

Reply:
Yes, because it's a macro. When used, the arguments are simply expanded as text, which can have unintended operator precedence consequences. For example, if we have:

    KextLog_VnodeOp(vn, isDir ? VDIR : VREG, ...);

The `if` statement expands to:

   if (VDIR == isDir ? VDIR : VREG)

Which is equivalent to:

   if ((VDIR == isDir) ? VDIR : VREG)

or

   if (true)

because both `VDIR` and `VREG` are non-zero. This is basically never what the user of the macro wanted!